apps/spark/src/main/java/com/linkedin/openhouse/jobs/util/AppsOtelEmitter.java

```diff
@@ -0,0 +1,105 @@
+package com.linkedin.openhouse.jobs.util;
+
+import com.linkedin.openhouse.common.OtelEmitter;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.LongGaugeBuilder;
+import io.opentelemetry.api.metrics.LongHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public final class AppsOtelEmitter implements OtelEmitter {
+  private final List<OpenTelemetry> otels;
+  private static volatile AppsOtelEmitter instance;
+
+  private AppsOtelEmitter(List<OpenTelemetry> otels) {
+    this.otels = otels;
+  }
+
+  public static synchronized AppsOtelEmitter getInstance() {
```

Comment:
Is the plan here to override the AppsOtelEmitter in li repo? How to we get li specific instance?

Reply:
We will supply a different OtelEmitter in li-repo. You can see that this is passed during apps initialization. We will need to customize all apps in li-repo.